tests/data/meta_tensor/test_meta_tensor.py

```diff
@@ -32,13 +32,14 @@
 from monai.data.utils import decollate_batch, list_data_collate
 from monai.transforms import BorderPadd, Compose, DivisiblePadd, FromMetaTensord, ToMetaTensord
 from monai.utils.enums import PostFix
-from tests.test_utils import TEST_DEVICES, SkipIfBeforePyTorchVersion, assert_allclose, skip_if_no_cuda
+from tests.test_utils import TEST_DEVICES, SkipIfBeforePyTorchVersion, assert_allclose, dict_product, skip_if_no_cuda
 
 DTYPES = [[torch.float32], [torch.float64], [torch.float16], [torch.int64], [torch.int32], [None]]
+
+# Replace nested loops with dict_product
 TESTS = []
-for _device in TEST_DEVICES:
-    for _dtype in DTYPES:
-        TESTS.append((*_device, *_dtype))  # type: ignore
+for params in dict_product(device=TEST_DEVICES, dtype=DTYPES):
+    TESTS.append((*params["device"], *params["dtype"]))  # type: ignore
```

Comment:
```suggestion
TESTS = [(*p["device"], *p["dtype"]) for p in dict_product(device=TEST_DEVICES, dtype=DTYPES)] # type: ignore
```
In some places a list comprehension may be fine to use, and not have to have `TESTS = []` before.